### DISCUSSION TITLE
4차 스프린트 개발 목록에 대해

### DISCUSSION THREAD

**Discussion by Choi-JJunho** [woowacourse-teams/2023-pium#194] [뜨거운 토론]
## 주제

cc. @WaiNaat @hozzijeong @bassyu @rawfishthelgh @Kim0914 @yeonkkk @Choi-JJunho 

> 본 discussion에서 다룰 내용은 프론트/백엔드의 기술적인 논의가 아닌 기획적인 측면에서 협의 내용입니다.
로깅, 시맨틱 버저닝 등 각 트랙별로 논의되어야하는 내용은 별도의 discussion을 참조해주세요

피움 개발의 마지막 데모데이 4차 데모데이를 진행하기에 앞서 기능 및 작업 목록을 명확하게 가져가봅시다.

### 3차 데모데이 피드백 정리

- 식물을 등록하는 과정에서 사전 식물에 대한 정보를 제공해주면 좋겠다.
- 등록하는 화면에서 뒤로가기 버튼이 존재하면 좋겠다.
- 리마인더 달력에서 백드롭 클릭했을 때 캘린더 내려가도록 구성했으면 좋겠다.
- 리마인더 달력에서 오늘 날짜에 커서가 올라가있는 경우를 생각해보면 좋겠다.
  - 오늘 물을 줄 수 없는데 빨간색 커서가 올라가있으면 사용자가 혼란스러워한다.
- 물주기 타임라인에 대해 물주기 기록 외 다른 기록도 보여주면 좋겠다.
  - 사용자가 등록한 날짜
  - 사용자가 수정한 날짜 및 정보
  - 물주기 정보 등
- DB Table Drop 제한에 대해 prod / dev 환경을 구분하라는 뜻이였다.
- 리마인더, 내 식물 보기에서 페이징 적용

### (필수) 기능 개발 목록

- 캘린더 기능
  - https://github.com/woowacourse-teams/2023-pium/discussions/205
- 로그인 기능
  - [x] API 명세
  - OAuth
    - [x] Kakao
- 식물 삭제기능
  - [x] API 명세

### (선택) 기능 개발 목록

- 식물 무덤관련기능
  - 식물 만료기능
  - 만료된 식물 조회 기능
- 리마인더 취소 기능
- 반려 식물 사진 등록 및 수정

**Comment by yeonkkk** [comment]
리마인더 알림 기능에 대한 논의도 하면 좋을 것 같아요!

**Reply by WaiNaat** [reply]
필수적인 기능은 아닌 것 같아 일단 보류하는 방향으로 결정되었습니다

**Comment by Choi-JJunho** [comment]
## 로그인 기능

일반적인 유저 입장에서 가장 접근성이 높은 카카오 로그인으로 결정한다.

**Reply by rawfishthelgh** [reply]
로그인 방식을 세션 방식으로 진행하는 것 vs 토큰 방식으로 진행하는 것
-> 서버의 부하나 유저의 증가는 아직 우리가 직면하지 않은 문제라고 판단함. 따라서 토큰 기반 방식을 사용하는 것은 개발자의 욕심일 수 있고, 불필요한 리소스를 사용하는 것 같다.
결론 : 세션 기반으로 로그인 기능을 구현한다.

**Comment by Choi-JJunho** [comment]
2023.08.08(화) 회의 결과 다음과 같은 결론으로 해당 논의 완료되었습니다.

## 타임라인 (반려 식물 히스토리 단건 조회)

- https://github.com/woowacourse-teams/2023-pium/discussions/205

해당 discussion 결과 캘린더 기능 대신 히스토리 기능 고도화로 결론 지어졌습니다.

따라서 `/history` API의 명세가 다음과 같이 변경되었습니다.

### API 명세

```json
{
  "page": 1,
  "size": 10,
  "elementSize": 100,
  "hasNext": true,
  "data": [
    {
      "type": "lastWaterDate",
      "date": "2023-07-01",
      "content": {
        "prev": null, // 첫 등록 시 기록은 prev를 null로 한다
        "current": "2023-07-01" // lastWaterDate의 prev -> current 이렇게 바꼈어용
      }
    },
    {
    "type": "waterCycle"
    "date": "2023-07-07",
    "content": {
        "prev": "3",
        "current": "5"
      }
    },
    {
      "type": "flowerpot"
      "date": "2023-07-07",
      "content": {
        "prev": "플라스틱 화분",
        "current": "유리 화분"
      }
    },
    {
      "type": "light"
      "date": "2023-07-07",
      "content": {
        "prev": "채광 잘되는곳",
        "current": "채광 안되는곳"
      }
    }
  ]
}
```

## 버그리포팅

버그리포팅 활동 간 작성된 이슈에 대한 협의내용입니다.

### 단어 선택

> 연관이슈
- #201 

반려 식물 정보 수정 페이지에서 다음과 같은 단어로 변경되었습니다.

- `생일` -> `입양일`
- `피움과 함께한 지` -> `함께한 지`

### 물주기 

> 연관 이슈
- #197 

기존에는 `타임라인 상에서 물주기 버튼을 눌렀을 때 증가하는 주기`로 인지하여 물 주기 값에 대한 변경은 독립적이다. 라고 기획되어있었습니다.

![image](https://github.com/woowacourse-teams/2023-pium/assets/49794401/df4a59dc-a241-496a-a245-b9b3ef09049f)

위 상황에서 사용자의 오해를 발생시킨다는 결론으로 
`물주기 날짜를 수정 시 다음 물주기 날짜를 변경한다.` 로 변경하는 것으로 결론 지었습니다.
